### PR TITLE
feat(ui): Add icon prop to tag component

### PR DESCRIPTION
**Pull request by evanpurkhiser**
[Story in storybook](https://storybook.getsentry.net/branches/featui-add-icon-prop-to-tag-component/index.html?selectedKind=UI%7CTags&selectedStory=with%20icon&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel)

image

### CODE REVIEW

**Review by markstory — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/components/tag.jsx [hunk 1]
```diff
@@ -43,6 +46,11 @@ Tag.propTypes = {
   priority: PropTypes.string,
   size: PropTypes.string,
   border: PropTypes.bool,
+  icon: PropTypes.string,
 };
 
+const StyledInlineSvg = styled(InlineSvg)`
+  margin-right: 4px;
+`;
```
[the remark this answers is not in the document]
Comment (EvanPurkhiser) [reply src/sentry/static/sentry/app/views/settings/components/tag.jsx]: Yeah we do. I'm not sure what our general consensus around padding / margin properties on components is these days, but maybe we would want to add `p` and `m` props to this component? /cc @billyvg @Chrissy @ckj

[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/tag.jsx]: I would like to get away from `p` and `m` props and would either prefer a good default or a more semantic prop name.

[the remark this answers is not in the document]
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/components/tag.jsx]: Do we need a configurable prop? A second export with a fixed right margin would cover several use cases.

[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/components/tag.jsx]: Yeah I think an additional export would be good if we find ourselves using this style a lot.

**Review by billyvg — APPROVED** [review]

**Pull request merged**